storage/database/sharded_database.go

```diff
@@ -151,102 +153,188 @@ func (pdb *shardedDB) Close() {
 	}
 }
 
+// Not enough size of channel slows down the iterator
+const shardedDBCombineChanSize = 1024 // Size of resultCh
+const shardedDBSubChannelSize = 128   // Size of each channel of resultChs
+
+// shardedDBIterator iterates all items of each shardDB.
+// This is useful when you want to get items in serial.
 type shardedDBIterator struct {
-	// TODO-Klaytn implement this later.
-	iterators []Iterator
-	key       []byte
-	value     []byte
+	shardedDBChanIterator
 
-	//numBatches uint
-	//
-	//taskCh   chan pdbBatchTask
-	//resultCh chan pdbBatchResult
+	resultCh chan entry
+	key      []byte // current key
+	value    []byte // current value
 }
 
-// NewIterator creates a binary-alphabetical iterator over the entire keyspace
-// contained within the key-value database.
+// NewIterator creates a iterator over the entire keyspace contained within
+// the key-value database.
+// If you want to get items in parallel from channels, checkout shardedDB.NewChanIterator()
 func (pdb *shardedDB) NewIterator() Iterator {
-	// TODO-Klaytn implement this later.
-	return nil
+	return pdb.newIterator(func(db Database) Iterator { return db.NewIterator() })
```

Comment:
Could you rename `pdb` to `sdb` or `db`? `pdb` was originated from "partitionaedDB"

Reply:
I renamed it to `db` and `it` for functions related to the iterator. I will make another PR to change the names in other functions.